### ISSUE TITLE
Replication Errors with OPF

### ISSUE THREAD

**Issue by dh-pfsweb** [jephthai/OpenPasswordFilter#36]
Hello,

We installed OpenPasswordFilter recently and have had some problems with it.  We ran into the krbtgt error mentioned in an older post.  In addition to that, we are having replication issues between our domains.  When we change a users password, or make other attribute changes, we need to stop the OPF Service on the DC to get the changes to replicate to the other DCs.  Has anyone else run into this issue or have any suggestions as to what we might be able to do to correct this problem?

thanks,
david

**Comment by FFFreak** [comment]
Yes, if you have a cloud sync it MUST be the last ones in the LSA
notifiers.  Especially Gsync (DEAD LAST!) which likes to kill the stack..
make sure OPF is before them, especially GSYNC!

edited:
And that is on **ALL** ADs, but can test on one that your changing the PW on - using powershell with -server <ip/hostname> switch

OPF shouldnt affect attributes though, so could also be something else.

**Comment by FFFreak** [comment]
And that is on **ALL** ADs, but can test on one that your changing the PW on - using powershell with -server <ip/hostname> switch

**Comment by dh-pfsweb** [comment]
Is this what you are referring to?

[cid:image003.png@01D5FDFA.929F6A70]

Just moving the OpenPasswordFilter to the top of the Notification Packages in the registry?

thanks,

david

**Comment by FFFreak** [comment]
Cannot see image, but I did it on mine before the cloud providers.  However, most things "pass-through" so i don't know of any issue with it being first, but i also have never tried it that way.  Sorry i cannot tell you a yes or no on it.

**Comment by dh-pfsweb** [comment]
Sorry.  The image was a screenshot of our registry.  I was wondering if this was the proper place to make the change to the LSA Notification.

HKLM > SYSTEM > CurrentControlSet > Control > Lsa > Notification Packages

In there I find three values:
               rassfm
scecli
OpenPasswordFilter

Do I just need to move OpenPasswordFilter to the top?

thanks,

david

**Comment by FFFreak** [comment]
I personally see no issue with the original order.  When you remove (OPF) it does replication perfectly? (replication and partition syncing [like on a service restart] are very different).

**Comment by FFFreak** [comment]
Also what Operating System (I'm just some dude, and played alot with OPF in my test environments, but OSes I think were 2008 R2 to 2012).

**Comment by dh-pfsweb** [comment]
This is installed on Server 2012 R2.

All we have to do is to disable the OPF Service on the DCs and everything works fine.  With the service enabled, we have long delays to change passwords (20 seconds up to several minutes), and we see replication problems arise.


thanks,

david

**Comment by FFFreak** [comment]
So OPF does use a sorta local loopback (127.0.0.1) to do the communication from LSA notifier to the service that does the checking.  Have you tried wire shark to see if there is a communication issue with this loopback.  I am wondering if that communication channel is having issues and your hitting a timeout on the call.  I didn't write it, but figured they did this for a buffer and ability to queue up asynchronous calls in to a synchronous check.
